src/prime/prime.cpp

```diff
@@ -452,6 +456,19 @@ bool CheckPrimeProofOfWork(uint256 hashBlockHeader, unsigned int nBits, const CB
     unsigned int nChainLengthCunningham2FermatTest = 0;
     unsigned int nChainLengthBiTwinFermatTest = 0;
     if (!ProbablePrimeChainTest(bnPrimeChainOrigin, nBits, true, nChainLengthCunningham1FermatTest, nChainLengthCunningham2FermatTest, nChainLengthBiTwinFermatTest)) {
+        // Despite failing the check, still return info of longest primechain from the three chain types
+        nChainLength = nChainLengthCunningham1;
+        nChainType = PRIME_CHAIN_CUNNINGHAM1;
+        if (nChainLengthCunningham2 > nChainLength)
+        {
+            nChainLength = nChainLengthCunningham2;
+            nChainType = PRIME_CHAIN_CUNNINGHAM2;
+        }
+        if (nChainLengthBiTwin > nChainLength)
+        {
+            nChainLength = nChainLengthBiTwin;
+            nChainType = PRIME_CHAIN_BI_TWIN;
+        }
         LogPrint(BCLog::PRIME, "CheckPrimeProofOfWork() : failed Fermat test target=%s length=(%s %s %s) lengthFermat=(%s %s %s)", TargetToString(nBits).c_str(),

```

Comment:
This block should be at line 445 instead. This second check fail condition is probably not reachable